### PR TITLE
consistently use ParallelCommunication.hpp for communication definition

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclmpiserializer.hh [hunk 1]
```diff
@@ -21,7 +21,6 @@
 #ifndef ECL_MPI_SERIALIZER_HH
 #define ECL_MPI_SERIALIZER_HH
 
-#include <dune/common/version.hh>
 #include <opm/simulators/utils/ParallelRestart.hpp>
```
Comment (bska) [inline ebos/eclmpiserializer.hh:24]: Does this implicitly rely on `ParallelRestart.hpp` also bringing in `ParallelCommunication.hpp`?  If so, I think the requirement  should be made explicit by including `ParallelCommunication.hpp` directly.

File: opm/simulators/wells/WellState.cpp [hunk 2]
```diff
@@ -825,8 +825,8 @@ WellState::parallelWellInfo(std::size_t well_index) const
     return ws.parallel_info;
 }
 
-template void WellState::updateGlobalIsGrup<ParallelWellInfo::Communication>(const ParallelWellInfo::Communication& comm);
-template void WellState::communicateGroupRates<ParallelWellInfo::Communication>(const ParallelWellInfo::Communication& comm);
+template void WellState::updateGlobalIsGrup<Parallel::Communication>(const Parallel::Communication& comm);
+template void WellState::communicateGroupRates<Parallel::Communication>(const Parallel::Communication& comm);
```
Comment (bska) [inline opm/simulators/wells/WellState.cpp:829]: This appears to rely on `ParallelCommunication.hpp` being brought in through transitive includes.  I'd recommend adding the header directly.

File: tests/test_parallelwellinfo.cpp [hunk 3]
```diff
@@ -118,7 +118,7 @@ BOOST_AUTO_TEST_CASE(ParallelWellComparison)
     std::vector<Opm::ParallelWellInfo> well_info;
     
     for (const auto& wellinfo : pairs) {                   
-        well_info.emplace_back(wellinfo, Opm::ParallelWellInfo::Communication());         
+        well_info.emplace_back(wellinfo, Opm::Parallel::Communication());
```
Comment (bska) [inline tests/test_parallelwellinfo.cpp:121]: Should probably include `ParallelCommunication.hpp` directly then.

**Review by bska — COMMENTED** [review]
This looks good other than a couple of places that appear to rely on `ParallelCommunication.hpp` being brought in through transitive includes.  Reducing the number of ways of defining our communicators can only be a good thing.

File: tests/test_parallelwellinfo.cpp (hunk 3 above)
Comment (akva2) [reply tests/test_parallelwellinfo.cpp:121]: done

File: opm/simulators/wells/WellState.cpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/wells/WellState.cpp:831]: done

File: ebos/eclmpiserializer.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclmpiserializer.hh]: done

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**